### PR TITLE
Fixes

### LINKED ISSUE #159: Ruff autoupdate on creating a PR which makes mandatory a previously optional rule
## 🐛 Bug Report

ruff updates break the CI by making rules that were previously optional mandatory. 
The rule is "PLR0917", # too many positional arguments. 

An easy fix is just adding that to the ignore part of the pyproject.toml (which I have done on my branches) but we may want to have a more long-term solution.

## 🔬 How To Reproduce

1. Run a PR on your current branch and you will have a Run style check build failure

### Code sample

The problem comes from workflows/build.yml

    - name: Install dependencies
      run: |
        source .venv/bin/activate
        make update-deps
        uv pip install -r requirements-dev.txt

which calls 

update-deps:
	uv pip compile pyproject.toml -o requirements.txt
	uv pip compile --all-extras pyproject.toml -o requirements-dev.txt

which compiles pyproject.toml namely 

lint = [
  "setuptools",
  "pydoclint>=0.4.0",
  "pre-commit>=2.19.0",
  "pytest>=7.2.0",
  "pytest-cov>=4.0.0",
  "pytest-xdist>=3.5.0",
  "ruff>=0.2.0",
  "virtualenv>=20.26.6",
  "networkx>=3.0.0",
  "numpy>=2.2.0",
]

so it defaults to the newest version of ruff, which clashes with the actual rules of the repo 

## Expected behavior

a CI that works correctly since I ran the linter 

## Additional context

We should maybe bound the ruff version ?

### PR DESCRIPTION
**Pull request by AntoninPoche**
## Description

Solves a collection of issues:
- Add `forward_kwargs` in splitters to allow specifying `truncation=True`.
- In some weird case, Shap was called with a `mask_dim=0`. It now supports this.
- Clean all error messages and warnings from the documentation compilation:
  - Treat broken links
  - Update some docstrings
  - Remove empty changelogs
  - Force `mkdocs==1.x`
  - Build doc only once
- Fix issue #159 by fixing `ruff==0.16.4`
- Fix notebook rendering errors in the doc by migrating from `mknotebooks` to `mkdocs-jupyter`
- Concept interpretations support `use_unique_words` with `SAMPLE` granularity

## Related Issue

#159 

## Type of Change

- 📚 Examples / docs / tutorials / dependencies update
- 🔧 Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/docs/contributing.md) guide.
- [x] I've successfully run the style checks using `make lint`.
- [x] I've written tests for all new methods and classes that I created and successfully ran `make test`.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: interpreto/concepts/interpretations/base.py [hunk 1]
```diff
@@ -534,15 +539,21 @@ def get_granular_inputs_and_concept_activations(
                 # ----------------------------------------------------------------------------------
                 # Case 2: use_unique_words >= 1
                 # first list unique words/ngrams from the inputs and compute the activations from them
-                if self.activation_granularity != ActivationGranularity.CLS_TOKEN:
+                if self.activation_granularity not in [
+                    ActivationGranularity.CLS_TOKEN,
+                    ActivationGranularity.SAMPLE,
+                ]:
```
[the remark this answers is not in the document]
Comment (AntoninPoche) [reply interpreto/concepts/interpretations/base.py:545]: This is just for the `ModelWithSplitPoints`.

People using `SplitterForGeneration` and `SplitterForClassification` should no longer set granularities. This simplifies the API.

File: interpreto/attributions/perturbations/shap_perturbation.py [hunk 2]
```diff
@@ -94,6 +94,9 @@ def get_mask(self, mask_dim: int) -> Float[Tensor, "p {mask_dim}"]:
         # Simplify typing
         p, l = self.n_perturbations, mask_dim
 
+        if l < 1:
+            return torch.zeros((p, l), dtype=torch.float)
```
[the remark this answers is not in the document]
Comment (AntoninPoche) [reply interpreto/attributions/perturbations/shap_perturbation.py:98]: Was solved.

[the remark this answers is not in the document]
Comment (AntoninPoche) [reply interpreto/attributions/perturbations/shap_perturbation.py]: Furthermore, these 0-dim inputs are filtered at some point and do not reach the aggregation.

**Review by fanny-jourdan — APPROVED** [review]
All right, Antonin. I don't really have any comments—everything looks good to me. Good job !

File: interpreto/concepts/interpretations/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntoninPoche) [reply interpreto/concepts/interpretations/base.py:545]: This is a comment on a deprecated API that will be removed in the next minor. Not worth the time.
